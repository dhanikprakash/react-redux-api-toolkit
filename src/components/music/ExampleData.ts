import { MusicBO } from "../../BO/music";

export const exampleData: MusicBO[] = [
  {
    userId: 1,
    id: 1580652352,
    title: "Lil Wayne",
    body: "Tha Carter IV (Complete Edition)",
  },
];

export const initialData: MusicBO[] = [
  {
    userId: 0,
    id: 0,
    title: "",
    body: "",
  },
];
