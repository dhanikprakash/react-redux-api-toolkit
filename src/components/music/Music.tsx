import React from "react";
import { useSelector } from "react-redux";

import { MusicSection } from "./MusicSection";
import { musicResults } from "./MusicSelector";

export const Music: React.FC = () => {
  const results = useSelector(musicResults);
  return (
    <MusicSection
      searchResults={results.searchResults}
      status={results.status}
    />
  );
};
