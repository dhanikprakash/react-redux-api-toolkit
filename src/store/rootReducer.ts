import { configureStore } from '@reduxjs/toolkit';
import musicReducer from '../components/music/MusicSlice';

export const store = configureStore({
  reducer: {
    musicResults: musicReducer,
  },
})

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch