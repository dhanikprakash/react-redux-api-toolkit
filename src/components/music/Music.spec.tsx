import { Provider } from "react-redux";
import { store } from "../../store/rootReducer";
import MockAdapter from "axios-mock-adapter";
import { Music } from "./Music";
import axios from "axios";
import { exampleData } from "./ExampleData";
import * as renderer from "react-test-renderer";

describe("MusicSection", () => {
  let fetchDataMock: MockAdapter;

  beforeEach(() => {
    fetchDataMock = new MockAdapter(axios);
    fetchDataMock
      .onGet("https://jsonplaceholder.typicode.com/posts")
      .reply(200, exampleData);
  });

  afterEach(() => {
    fetchDataMock.restore();
  });

  test("should render", async () => {
    const data = renderer
      .create(
        <Provider store={store}>
          <Music />
        </Provider>
      )
      .toJSON();

    expect(data).toMatchSnapshot();
  });
});
