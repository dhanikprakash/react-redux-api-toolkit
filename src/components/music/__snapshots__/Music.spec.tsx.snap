// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicSection should render 1`] = `
<div
  className="flex-container"
>
  <div
    className="music-search"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 css-1qz17ui-MuiPaper-root"
      style={
        Object {
          "padding": 30,
        }
      }
    >
      <h3>
        Search
      </h3>
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="music-search"
          id="music-search-label"
        >
          Music / Song / Artist
        </label>
        <div
          className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled={false}
            id="music-search"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Search
      </button>
    </div>
  </div>
  <div
    className="music-result"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 css-1qz17ui-MuiPaper-root"
      style={
        Object {
          "height": 500,
          "padding": 20,
        }
      }
    >
      <h3>
        Result
      </h3>
      <h4>
         
        idle
      </h4>
      <div />
    </div>
  </div>
</div>
`;
